### PR TITLE
AP_Camera: use of AP_Camera checks HAL_CAMERA_ENABLED

### PR DESCRIPTION
**Pull request by tatsuy**
I added HAL_CAMERA_ENABLED so that AP_Camera can be disabled.
I removed CAMERA and use only HAL_CAMERA_ENABLED.

This is CubeOrange build size.
```
Target          Text (B)  Data (B)  BSS (B)  Total Flash Used (B)  Free Flash (B)
---------------------------------------------------------------------------------
bin/arducopter   1776568      3172   259184               1779740          186328 // master (ab8c40d9)
bin/arducopter   1776568      3172   259184               1779740          186328 // HAL_CAMERA_ENABLED = 1
bin/arducopter   1772548      3172   259188               1775720          190344 // HAL_CAMERA_ENABLED = 0
```

In Sub, CAMERA was DISABLE by default, but this change will enable it.
I'm not sure if I should use the different default value according to vehicle types.

### CODE REVIEW

**Comment by tatsuy** [comment]
rebased

File: ArduCopter/Copter.cpp [hunk 1]
```diff
@@ -172,7 +172,7 @@ const AP_Scheduler::Task Copter::scheduler_tasks[] = {
 #if HAL_MOUNT_ENABLED
     SCHED_TASK_CLASS(AP_Mount,             &copter.camera_mount,        update,          50,  75, 108),
 #endif
-#if CAMERA == ENABLED
+#if HAL_CAMERA_ENABLED == ENABLED
```
Comment (peterbarker) [inline ArduCopter/Copter.cpp:175]:
```suggestion
#if AP_CAMERA_ENABLED
```

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: ArduCopter/Copter.cpp (hunk 1 above)
Comment (tatsuy) [reply ArduCopter/Copter.cpp]: fixed it

**Comment by tatsuy** [comment]
@peterbarker 
Thank you for your review!
I fixed.

**Comment by tatsuy** [comment]
Closed by #20891

**Pull request closed without merging**